reports/views.py

```diff
@@ -70,7 +73,19 @@ def report_content(request, permit_request_id, work_object_type_id, report_id):
 @login_required
 @permanent_user_required
 def report_pdf(request, permit_request_id, work_object_type_id, report_id):
+
+    # Check that user is trusted, as standard user is not alowed to generate documents
+    if not request.user.permitauthor.is_trusted(request.user):
+        raise Http404
+
+    # Prevent unauthorize user to call this view, return a 404 instead of the error from container
+    # This way, we avoid creating a container for nothing
+    get_object_or_404(
+        services.get_permit_requests_list_for_user(request.user), pk=permit_request_id
+    )
+
```

Comment:
A bit redundant because this will be checked by the report_content view anyway. But doesn't hurt either...

Arguably you could just test for `request.user.has_perm("reports.can_generate_pdf")` ?